api/config/rolepermission.php

```diff
@@ -0,0 +1,521 @@
+            'fr' => 'Archiver sa propre demande'
+        ],
+        'view-any-applicantCount' => [
+            'en' => 'View Applicant Count',
+            'fr' => 'Voir le nombre de candidats'
+        ],
+
+        'create-any-searchRequest' => [
+            'en' => 'Create Any Search Request',
+            'fr' => 'Créer toute demande de recherche'
+        ],
+        'view-team-searchRequest' => [
+            'en' => 'Create Team Search Request',
+            'fr' => 'Créer une demande de recherche d\'équipe'
+        ],
+        'update-team-searchRequest' => [
+            'en' => 'Update Team Search Request',
+            'fr' => 'Mise à jour de la demande de recherche d\'équipe'
+        ],
+        'delete-team-searchRequest' => [
+            'en' => 'Delete Team Search Request',
+            'fr' => 'Supprimer une demande de recherche d\'équipe'
+        ],
+
+        'view-any-team' => [
+            'en' => 'View Any Team',
+            'fr' => 'Voir n\'importe quelle équipe'
+        ],
+        'view-any-teamUsersBasicInfo' => [
+            'en' => 'View Any Team Users Basic Info',
+            'fr' => 'Afficher les informations de base sur les utilisateurs de n\'importe quelle équipe'
+        ],
+        'view-team-teamUsersBasicInfo' => [
+            'en' => 'View Teams Team Users Basic Info',
+            'fr' => 'Afficher les informations de base sur les utilisateurs de l\'équipe'
+        ],
+        'create-any-team' => [
+            'en' => 'Create Any Team',
+            'fr' => 'Créer n\'importe quelle équipe'
+        ],
+        'update-any-team' => [
+            'en' => 'Update Any Team',
+            'fr' => 'Mise à jour de n\'importe quelle équipe'
+        ],
+        'update-team-team' => [
+            'en' => 'Update Team Team',
+            'fr' => 'Mise à jour de l\'équipe'
+        ],
+        'delete-any-team' => [
+            'en' => 'Delete Any Team',
+            'fr' => 'Supprimer une équipe'
+        ],
+
+        'view-any-role' => [
+            'en' => 'View Any Role',
+            'fr' => 'Voir tous les rôles'
+        ],
+        'assign-any-role' => [
+            'en' => 'Assign Any Role',
+            'fr' => 'Attribuer n\'importe quel rôle'
+        ],
+        'assign-team-role' => [
+            'en' => 'Assign Team Role',
+            'fr' => 'Attribuer un rôle à l\'équipe'
+        ],
+        'update-any-role' => [
+            'en' => 'Update Any Role',
+            'fr' => 'Mise à jour de n\'importe quel rôle'
+        ],
+    ],
+
+    /*
+    |--------------------------------------------------------------------------
+    | Roles
+    |--------------------------------------------------------------------------
+    |
+    | A map of all roles with the shape:
+    |
+    |   $name => [
+    |       'display_name' => LocalizedString,
+    |       'description' => LocalizedString,
+    |   ]
+    |
+    */
+    'roles' => [
+        'guest' => [
```

Comment:
How could someone have any role if they are not logged in?

Reply:
I may be wrong on this, but this role is supposed to act like a façade to assign anonymous permissions, if the current user is null, we use this role to determine what they can do. 

We are also not using permission inheritance, rather assigning multiple roles to add permissions. So a user would have:

 - Guest
 - Base User
 - Applicant

And all permissions that come with that.

Reply:
Possibly like this:
```
if ($user) {
  return $user->isAbleTo('view-something');
} else {
  Role::where('name', 'guest')->first()->hasPermission('view-something');
}
```
And if not that, then just for organizing and recording our own understanding of permissions.

Reply:
1. Is it likely that users will require multiple roles and if so, why?
2. Is there a scenario where having multiple roles that contain conflicting permissions could occur?

Reply:
1.Yes. For example, it is likely that a user with a high-level role (eg for creating teams) may also be able to act as a lower-level role (eg for creating pools in a team). As an alternative, we _could_ design high-level roles to be strictly a superset of low-level roles (all the same permissions, plus more) and give every user only one role, but that is more restrictive and less flexible.
2. Having more roles gives you more permissions in a purely additive way, so I don't think there's any way they could conflict.